core/src/main/java/com/netflix/conductor/core/dal/ExecutionDAOFacade.java

```diff
@@ -331,20 +331,43 @@ public void removeFromPendingWorkflow(String workflowType, String workflowId) {
     /**
      * Removes the workflow from the data store.
      *
-     * @param workflowId the id of the workflow to be removed
+     * @param workflowId the workflow to be removed
      * @param archiveWorkflow if true, the workflow will be archived in the {@link IndexDAO} after
      *     removal from {@link ExecutionDAO}
+     * @param removeTasks if true, the workflow associated tasks of the workflow will be removed
+     * @param archiveTasks if true, the workflow associated tasks will be archived in the {@link IndexDAO} after
+     *     removal from {@link ExecutionDAO}
      */
-    public void removeWorkflow(String workflowId, boolean archiveWorkflow) {
+    public void removeWorkflow(String workflowId, boolean archiveWorkflow, boolean removeTasks, boolean archiveTasks) {
```

Comment:
I'm not sure I agree that we need to do this kind of extra task logic in the API here. I think that we should just either archive tasks or not archive tasks, in conjunction with the workflow. I think it would be weird to archive one or the other IMO.

Reply:
I agree, but let's propose this to the guys over at Netflix as I don't want to introduce unexpected/implicit behaviour. Either the implicit behaviour should be documented or the option should be present.